From e8f5e6f539b3f60ca362bfa52c0b719442307cae Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sat, 9 Apr 2022 20:15:53 +0200
Subject: [PATCH 013/172] Remove unnecessary macro

allocStr returns NULL if the passed pointer is NULL, no need to use a
NULL check here.
---
 url.c |   16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

--- a/url.c
+++ b/url.c
@@ -1046,8 +1046,6 @@ parseURL(char *url, ParsedURL *p_url, Pa
 	p_url->label = NULL;
 }
 
-#define ALLOC_STR(s) ((s)==NULL?NULL:allocStr(s,-1))
-
 void
 copyParsedURL(ParsedURL *p, const ParsedURL *q)
 {
@@ -1059,13 +1057,13 @@ copyParsedURL(ParsedURL *p, const Parsed
     p->scheme = q->scheme;
     p->port = q->port;
     p->is_nocache = q->is_nocache;
-    p->user = ALLOC_STR(q->user);
-    p->pass = ALLOC_STR(q->pass);
-    p->host = ALLOC_STR(q->host);
-    p->file = ALLOC_STR(q->file);
-    p->real_file = ALLOC_STR(q->real_file);
-    p->label = ALLOC_STR(q->label);
-    p->query = ALLOC_STR(q->query);
+    p->user = allocStr(q->user, -1);
+    p->pass = allocStr(q->pass, -1);
+    p->host = allocStr(q->host, -1);
+    p->file = allocStr(q->file, -1);
+    p->real_file = allocStr(q->real_file, -1);
+    p->label = allocStr(q->label, -1);
+    p->query = allocStr(q->query, -1);
 }
 
 void
